mono/metadata/class-internals.h

```diff
@@ -1234,6 +1255,21 @@ mono_create_icall_signature (const char *sigstr);
 MonoJitICallInfo *
 mono_register_jit_icall (gconstpointer func, const char *name, MonoMethodSignature *sig, gboolean is_save);
 
+#ifdef __cplusplus
+
+extern "C++" { // in case wrapped in extern "C"
+
+template <typename T>
+inline MonoJitICallInfo*
```

Comment:
Yeah regular inline, not extern, not static, very likely to be inlined, but linking succeeds either way.